src/yunohost/app.py

```diff
@@ -341,6 +341,105 @@ def app_map(app=None, raw=False, user=None):
+                              m18n.n('app_not_installed', app=app))
+
+    # checkurl will modify the settings
+    # (this is a non intuitive behavior that should be changed)
+    # (or checkurl renamed in reserve_url)
+    old_domain = app_setting(app, "domain")
+    old_path = app_setting(app, "path")
+
+    old_path_trimed = old_path.rstrip("/")
+
+    if (domain.strip().lower(), path.rstrip("/").strip()) == (old_domain, old_path_trimed.strip()):
+        raise MoulinetteError(errno.EINVAL, m18n.n("app_change_url_identical_domains", domain=domain, path=path))
+
+    app_checkurl(auth, '%s%s' % (domain, path), app)
+
+    if not os.path.exists(os.path.join(APPS_SETTING_PATH, app, "scripts", "change_url")):
+        raise MoulinetteError(errno.EINVAL, m18n.n("app_change_no_change_url_script", app_name=app))
+
+    manifest = json.load(open(os.path.join(APPS_SETTING_PATH, app, "manifest.json")))
+
+    # Retrieve arguments list for change_url script
+    # TODO: Allow to specify arguments
+    args_odict = _parse_args_from_manifest(manifest, 'change_url', auth=auth)
+    args_list = args_odict.values()
+    args_list.append(app)
+
+    # Prepare env. var. to pass to script
+    env_dict = _make_environment_dict(args_odict)
+    app_id, app_instance_nb = _parse_app_instance_name(app)
+    env_dict["YNH_APP_ID"] = app_id
+    env_dict["YNH_APP_INSTANCE_NAME"] = app
+    env_dict["YNH_APP_INSTANCE_NUMBER"] = str(app_instance_nb)
+
+    env_dict["YNH_APP_OLD_DOMAIN"] = old_domain
+    env_dict["YNH_APP_OLD_PATH"] = old_path
+    env_dict["YNH_APP_OLD_PATH_TRIMED"] = old_path_trimed
+
+    env_dict["YNH_APP_NEW_DOMAIN"] = domain
+    env_dict["YNH_APP_NEW_PATH"] = path
+
+    env_dict["YNH_APP_NEW_PATH_TRIMED"] = path.rstrip("/")
+
+    if os.path.exists(os.path.join(APP_TMP_FOLDER, "scripts")):
+        shutil.rmtree(os.path.join(APP_TMP_FOLDER, "scripts"))
+
+    shutil.copytree(os.path.join(APPS_SETTING_PATH, app, "scripts"),
+                    os.path.join(APP_TMP_FOLDER, "scripts"))
+
+    # Execute App change_url script
+    os.system('chown -R admin: %s' % INSTALL_TMP)
+    os.system('chmod +x %s' % os.path.join(os.path.join(APP_TMP_FOLDER, "scripts")))
+    os.system('chmod +x %s' % os.path.join(os.path.join(APP_TMP_FOLDER, "scripts", "change_url")))
+
+    # XXX journal
+    if hook_exec(os.path.join(APP_TMP_FOLDER, 'scripts/change_url'), args=args_list, env=env_dict) != 0:
+        logger.error("Failed to change '%s' url." % app)
+
+        # restore values modified by app_checkurl
+        # see begining of the function
+        app_setting(app, "domain", value=old_domain)
+        app_setting(app, "path", value=old_path)
+
+        return
+
+    # this should idealy be done in the change_url script but let's avoid common mistakes
+    app_setting(app, 'domain', value=domain)
+    app_setting(app, 'path', value=path)
+
+    app_ssowatconf(auth)
+
+    # avoid common mistakes
+    if os.system("service nginx reload") != 0:
```

Comment:
Shouldn't you use _run_service_command("reload", "nginx") from yunohost.service here ? Imho the whole thing with nginx -t should be factorized out because that's a recurring need (this kind of helper would be good to have for certificate.py for instance)